app/user.py

```diff
@@ -15,12 +15,36 @@ class AddSuccess:
     email: str
 
 
+@dataclass
+class AddError:
+    reason: Literal["CONFLICT", "UNKNOWN"]
+    message: str
+
+
 @dataclass
 class Error:
     reason: Literal["BAD_REQUEST", "CONFLICT", "UNKNOWN", "NOT_FOUND"]
     message: str
 
 
+@dataclass
```

Comment:
Instead of keeping every use case in the same module. You could transform user into a package and extract each use case to its own module. For example, a module for add_user ands its models, a module for update_user and its models, and so on.